File: ios/InvoiceApp/InvoiceApp/Domain/Models/Category.swift
import Foundation

struct Category: Codable, Identifiable, Hashable {
    let id: UUID
    let name: String
    let description: String?
    let icon: String?
    let color: String?
    let isDefault: Bool
    let userId: UUID?
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id, name, description, icon, color, isDefault, userId, createdAt, updatedAt
    }
}

struct CategoryCreateRequest: Codable {
    let name: String
    let description: String?
    let icon: String?
    let color: String?
}

struct CategoryUpdateRequest: Codable {
    let name: String?
    let description: String?
    let icon: String?
    let color: String?
}

struct CategoryStatistics: Codable {
    let categoryId: UUID
    let categoryName: String
    let totalAmount: Decimal
    let invoiceCount: Int
    let percentage: Double
}

File: android/app/src/main/java/com/invoiceapp/model/Category.kt
package com.invoiceapp.model

import java.time.LocalDateTime
import java.util.UUID

data class Category(
    val id: UUID,
    val name: String,
    val type: CategoryType,
    val icon: String?,
    val color: String?,
    val description: String?,
    val parentId: UUID?,
    val isCustom: Boolean,
    val userId: UUID?,
    val createdAt: LocalDateTime
)

enum class CategoryType {
    EXPENSE,
    INCOME
}
